### PR TITLE
EOS-26183: Filters support for Support bundle creation for Utils

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- Filters support for Support bundle creation for Utils

# Design
-  https://jts.seagate.com/browse/EOS-26183

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: Y
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: Y
- [x] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Is there a change in filename/package/module or signature [Y/N]: N
- [x] If yes for above point, Is a notification sent to all other cortx components [Y/N] NA
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/support/utils_support_bundle.py [hunk 1]
```diff
@@ -43,8 +43,10 @@ class UtilsSupportBundle:
     _tmp_src = '/tmp/cortx/py-utils/'
 
     @staticmethod
-    def generate(bundle_id: str, target_path: str, cluster_conf: str):
+    def generate(bundle_id: str, target_path: str, cluster_conf: str, duration: str,
+        size_limit: str, binlogs: bool, coredumps: bool, stacktrace: bool):
```
Comment (selvakumaar5496) [inline py-utils/src/support/utils_support_bundle.py:47]: Instead of increasing the no of params, Can you please have a kwargs and name it as **filters?

Comment (vastradparayya) [reply py-utils/src/support/utils_support_bundle.py]: Done

File: py-utils/src/support/utils_support_bundle.py [hunk 2]
```diff
@@ -111,16 +116,42 @@ def parse_args():
...
+        parser.add_argument('--coredumps', type=str2bool, default=False, dest='coredumps',
+            help="Include/Exclude Coredumps, Default = False")
+        parser.add_argument('--stacktrace', type=str2bool, default=False, dest='stacktrace',
+            help="Include/Exclude stacktrace, Default = False")
+        parser.add_argument('--modules', dest='modules',
+            help="list of components & services to generate support bundle.")
         args=parser.parse_args()
         return args
 
+def str2bool(value):
```
Comment (ujjwalpl) [inline py-utils/src/support/utils_support_bundle.py:134]: Do not add global functions.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/support/utils_support_bundle.py (hunk 2 above)
Comment (vastradparayya) [reply py-utils/src/support/utils_support_bundle.py]: Removed str2bool global method

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/support/utils_support_bundle.py [hunk 3]
```diff
@@ -111,16 +116,43 @@ def parse_args():
...
+    @staticmethod
+    def str2bool(value):
+        if isinstance(value, bool):
+            return value
+        if value.lower() in ('true'):
+            return True
+        elif value.lower() in ('false'):
+            return False
+        else:
+            raise argparse.ArgumentTypeError('Boolean value expected.')
```
Comment (ujjwalpl) [inline py-utils/src/support/utils_support_bundle.py:143]: Can you check and confirm passing an arbitrary value e.g. "foo" results in CORTX level exception e.g. SupportBundleError and does not lead to ArgumentTypeError exception to user unhandled ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/support/utils_support_bundle.py (hunk 3 above)
Comment (vastradparayya) [reply py-utils/src/support/utils_support_bundle.py:143]:
```
[root@ssc-vm-g3-rhev4-1094 cortx-utils]# utils_support_bundle -b SBag9s2g0355678_445c7c22dbaa4a3d978c0c98c3889c42_utils -t /var/log/cortx/support_bundle/SBag9s2g0355678/ -c yaml:///etc/cortx/cluster.conf -s all --duration P5D --size_limit 625.0MB --binlogs False --coredumps False **--stacktrace foo**
usage: utils_support_bundle [-h] [-b BUNDLE_ID] [-t [PATH]] [-c CLUSTER_CONF]
                            [-s SERVICES [SERVICES ...]] [-d DURATION]
                            [--size_limit SIZE_LIMIT] [--binlogs BINLOGS]
                            [--coredumps COREDUMPS] [--stacktrace STACKTRACE]
                            [--modules MODULES]
**utils_support_bundle: error: argument --stacktrace: Boolean value expected.**
```

**Review by ujjwalpl — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

**Comment by sachinpunadikar** [comment]
Ignoring codacy issues as the unused variables will be used in upcoming changes.
Ignoring whitesource issue as it was not added by this PR

**Pull request merged**